### ISSUE TITLE
Updating pattern (existing toolkits/solutions) aka Migration

### ISSUE THREAD

**Issue by jezzsantos** [jezzsantos/automate#25]
When you update a pattern and build it into the next version of the toolkit, AND you have the previous version of the toolkit installed, AND you have an existing solution created from previous version of the toolkit -> then:
1. The local installed toolkit is replaced by new version
2. The existing solution is not upgraded. The version is not upgraded, the pattern schema inside it is not upgraded, the codetemplatesfiles are not upgraded.

In order to use the updates to the pattern/toolkit, those things in the solution require upgrading, or else we need a different strategy for what is stored in a solution file itself.

It may be possible to upgrade automatically or manually (command: `automate upgrade solution`, but there is a chance that some data may be lost or changed (elements/collections/attributes/automation/codetemplatefiles/etc), and that will have to be accounted for in the data of the existing solution?

**Comment by jezzsantos** [comment]
Toolkit.json contains Pattern:
```
{
    "Version": "3.0.0",
    "PatternName": "BoostAPI",
	"Pattern": {...},
    "CodeTemplateFiles": [...],
    "Id": "05P610DZ"
}
```

Solution.json contains Toolkit and Pattern:
```
{
  "Toolkit": {
      "Version": "3.0.0",
      "PatternName": "BoostAPI",
	  "Pattern": {...},
      "CodeTemplateFiles": [...],
      "Id": "05P610DZ"
  },
    "PatternName": "BoostAPI",
    "Id": "JsaDBPUH",
    "Name": "BoostApi304",
    "Model": {...}
}
```

**Comment by jezzsantos** [comment]
This issue will need to consider the newer version of the toolkit, since now we have built-in breaking change detection.
A major toolkit upgrade may or may not be automatically upgradeable!!!

**Comment by jezzsantos** [comment]
Fixed by #32
